### DISCUSSION TITLE
Make Python available in front end.

### DISCUSSION THREAD

**Discussion by BobSquarePants** [brython-dev/brython#2214] [Ideas]
Brython is the way to go to make ( it's about time ) Python available for the front-end.

https://connect.mozilla.org/t5/ideas/gecko-firefox-supporting-python-to-interact-with-the-html-dom/idi-p/20158#M10542

On my side, I don't get why a web developer, should learn so many different languages in order to build and make-it available online a website ! here a random, short non-exhaustive list.

<hr>

Back-end
- Python (or other)
- SQL
- ~~HTTP server
- ~~firewall etc..
- ...

front-end
- html
- JavaScript
- CSS
- ...

~~= not per say a language, but rely on one to work, unless GUI.

<hr>

Depending on your case you might have even a lot more.

So being stuck to learn redundant languages to build one website it is not efficient !

especially JavaScript.. That in my point of view is at the opposite of Python , readable and concise !

On my side, I'm totally free of JavaScript coding ( thx to Brython ! ) 

Anyway Brython just show the way, but it could be good if Python was supported natively by Web Browser. How could we achieve that ?

@PierreQuentel , I've sent you an email did you received it ?

**Comment by PierreQuentel** [comment]
I agree 100% with you, Brython was designed to show how developers could write programs in Python instead of Javascript. The only way when I started the project (in 2012) was to translate Python code to Javascript; there is another option now, WASM, which is used by other projects, with pros and cons.

Native support of other languages than JS in browsers - that is, the ability to write client-side scripts in Python, Go, Rust, Julia, etc. by just specifying a type in the `<script>` tag without having to load a language-specific JS library such as **`brython.js`** - has been discussed for ages. It is the decision of browser vendors (Mozilla, Google, Microsoft, Apple etc.) and I see no sign of this happening in the near future. One of the arguments is that it would dramatically increase the size of browser downloads + add a maintenance burden, with responsabilities hard to define between the browser and the language developers.

**Reply by BobSquarePants** [reply]
> Brython was designed to show how developers could write programs in Python instead of Javascript.

Then it succeeded ! I've been fully convinced and I do not tape one line of JS anymore !!! Thx to have made Brython !!!
I've given few shot with WASM, but such headaches for now... (I've found nothing Pytonic yet.. )<br><br><br>



 >  It is the decision of browser vendors (Mozilla, Google, Microsoft, Apple etc.) and I see no sign of this happening in the near future. One of the arguments is that it would dramatically increase the size of browser downloads + add a maintenance burden, with responsabilities hard to define between the browser and the language developers.

Sadly, I witness the same, Maybe _we_ could develop an Browser (Firefox ?) add-on that will allow dev, to use `<script type="text/python">` !? and if it gain in popularity it may be merged directly in Firefox.. ( I saw this scenario in a couple of occasion )

And once again, Bravo for this amazing software that is [Brython ](https://brython.info/)!
